src/main/java/ma/hibernate/embeddable/dao/impl/NewsPostDaoImpl.java

```diff
@@ -11,11 +22,40 @@ public NewsPostDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public NewsPost save(NewsPost post) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
@EliosPride You use sessionFactory just 1 time, so I don't see any reason to save it to var